homeassistant/components/smartthings/media_player.py

```diff
@@ -0,0 +1,196 @@
+        self._state_attrs = None
+        self._supported_features = (
+            MediaPlayerEntityFeature.PLAY
+            | MediaPlayerEntityFeature.PAUSE
+            | MediaPlayerEntityFeature.STOP
+        )
+        if Capability.audio_volume in device.capabilities:
+            self._supported_features |= (
+                MediaPlayerEntityFeature.VOLUME_SET
+                | MediaPlayerEntityFeature.VOLUME_STEP
+            )
+        if Capability.audio_mute in device.capabilities:
+            self._supported_features |= MediaPlayerEntityFeature.VOLUME_MUTE
+        if Capability.switch in device.capabilities:
+            self._supported_features |= (
+                MediaPlayerEntityFeature.TURN_ON | MediaPlayerEntityFeature.TURN_OFF
+            )
+        if Capability.media_input_source in device.capabilities:
+            self._supported_features |= MediaPlayerEntityFeature.SELECT_SOURCE
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn the media player on."""
+        await self._device.switch_on(set_status=True)
+        self.async_write_ha_state()
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn the media player off."""
+        await self._device.switch_off(set_status=True)
+        self.async_write_ha_state()
+
+    async def async_mute_volume(self, mute: bool) -> None:
+        """Mute volume."""
+        if mute:
+            await self._device.mute(set_status=True)
+        else:
+            await self._device.unmute(set_status=True)
+        self.async_write_ha_state()
+
+    async def async_set_volume_level(self, volume: float) -> None:
+        """Set volume level."""
+        await self._device.set_volume(int(volume * 100), set_status=True)
+        self.async_write_ha_state()
+
+    async def async_volume_up(self) -> None:
+        """Increase volume."""
+        await self._device.volume_up(set_status=True)
+        self.async_write_ha_state()
+
+    async def async_volume_down(self) -> None:
+        """Decrease volume."""
+        await self._device.volume_down(set_status=True)
+        self.async_write_ha_state()
+
+    async def async_media_play(self) -> None:
+        """Play media."""
+        await self._device.play(set_status=True)
+        self.async_write_ha_state()
+
+    async def async_media_pause(self) -> None:
+        """Pause media."""
+        await self._device.pause(set_status=True)
+        self.async_write_ha_state()
+
+    async def async_media_stop(self) -> None:
+        """Stop media."""
+        await self._device.stop(set_status=True)
+        self.async_write_ha_state()
+
+    async def async_select_source(self, source: str) -> None:
+        """Select source."""
+        await self._device.set_input_source(source, set_status=True)
+        self.async_write_ha_state()
+
+    @property
+    def supported_features(self) -> MediaPlayerEntityFeature:
+        """Supported features."""
+        return self._supported_features
```

Comment:
Instead of returning it like this, rename `self._supported_features` to `self._attr_supported_features`, since the default implementation of the media player will automatcially use that

Reply:
Great, thanks !